src/Analyzer.Cli.FunctionalTests/Tests/TemplateWithModule.bicep

```diff
@@ -3,4 +3,11 @@ module testModule 'AppServicesLogs-Failures.bicep' = {
 	params: {
 		location: 'testLocation'
 	}
-}
\ No newline at end of file
+}
+
+module hw 'br/public:samples/hello-world:1.0.2' = {
```

Comment:
What is this reference?  Is this actually downloaded?  How does this impact the template analysis?

Reply:
This is the [bicep public module registry](https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/modules#public-module-registry), so it actually downloaded a bicep file and includes a resource in the resultant ARM template.

This was originally in the tests (I think this file was renamed?) to verify bicep files with modules did not have any issues, even though we did not process them. Now this is actually downloaded we should at least have a test to verify this works with no issues. I don't think validation beyond that is necessary. Let me know your thoughts here, I've added comments to clarify what this does in the template.